pkg/agent/controller/packetsampling/packetsampling_controller.go

```diff
@@ -0,0 +1,776 @@
+	nodeConfig *config.NodeConfig,
+	serviceCIDR *net.IPNet,
+	enableAntreaProxy bool,
+) *Controller {
+	c := &Controller{
+		kubeClient:             kubeClient,
+		crdClient:              crdClient,
+		packetSamplingInformer: packetSamplingInformer,
+		packetSamplingLister:   packetSamplingInformer.Lister(),
+		packetSamplingSynced:   packetSamplingInformer.Informer().HasSynced,
+		ofClient:               client,
+		interfaceStore:         interfaceStore,
+		networkConfig:          networkConfig,
+		nodeConfig:             nodeConfig,
+		serviceCIDR:            serviceCIDR,
+		queue:                  workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "packetsampling"),
+		runningPacketSamplings: make(map[uint8]*packetSamplingState),
+		sftpUploader:           &sftpUploader{},
+		enableAntreaProxy:      enableAntreaProxy,
+	}
+
+	packetSamplingInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.addPacketSampling,
+		UpdateFunc: c.updatePacketSampling,
+		DeleteFunc: c.deletePacketSampling,
+	}, resyncPeriod)
+
+	c.ofClient.RegisterPacketInHandler(uint8(openflow.PacketInCategoryPS), c)
+
+	if c.enableAntreaProxy {
+		c.serviceLister = serviceInformer.Lister()
+		c.serviceListerSynced = serviceInformer.Informer().HasSynced
+	}
+	return c
+
+}
+
+func (c *Controller) enqueuePacketSampling(ps *crdv1alpha1.PacketSampling) {
+	c.queue.Add(ps.Name)
+}
+
+// Run will create defaultWorkers workers (go routines) which will process the PacketSampling events from the
+// workqueue.
+func (c *Controller) Run(stopCh <-chan struct{}) {
+	defer c.queue.ShutDown()
+
+	klog.Infof("Starting %s", controllerName)
+	defer klog.Infof("Shutting down %s", controllerName)
+
+	cacheSynced := []cache.InformerSynced{c.packetSamplingSynced}
+	if c.enableAntreaProxy {
+		cacheSynced = append(cacheSynced, c.serviceListerSynced)
+	}
+	if !cache.WaitForNamedCacheSync(controllerName, stopCh, cacheSynced...) {
+		return
+	}
+
+	err := os.MkdirAll(packetDirectory, 0755)
+	if err != nil {
+		klog.ErrorS(err, "Couldn't create directory for storing sampling packets")
+		return
+	}
+
+	for i := 0; i < defaultWorkers; i++ {
+		go wait.Until(c.worker, time.Second, stopCh)
+	}
+	<-stopCh
+}
+
+func (c *Controller) addPacketSampling(obj interface{}) {
+	ps := obj.(*crdv1alpha1.PacketSampling)
+	klog.Infof("Processing PacketSampling %s ADD event", ps.Name)
+	c.enqueuePacketSampling(ps)
+}
+
+func (c *Controller) updatePacketSampling(_, obj interface{}) {
+	ps := obj.(*crdv1alpha1.PacketSampling)
+	klog.Infof("Processing PacketSampling %s UPDATE EVENT", ps.Name)
+	c.enqueuePacketSampling(ps)
+}
+
+func (c *Controller) deletePacketSampling(obj interface{}) {
+	ps := obj.(*crdv1alpha1.PacketSampling)
+	klog.Infof("Processing PacketSampling %s DELETE event", ps.Name)
+
+	err := deletePcapngFile(string(ps.UID))
+	if err != nil {
+		klog.ErrorS(err, "Couldn't delete pcapng file")
+
```

Comment:
```suggestion
		klog.ErrorS(err, "Couldn't delete pcapng file")
```
Please check and go through the codes, there are some unnecessary empty lines

Reply:
OK